compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/model/types/BType.java

```diff
@@ -162,6 +162,23 @@ public String getQualifiedTypeName() {
         return tsymbol.pkgID.toString() + ":" + tsymbol.name;
     }
 
+    public long getFlags() {
+        return flags;
+    }
+
+    public void setFlags(long flags) {
```

Comment:
Would it be better if we simply override this method in child classes (`BMapType`, `BRecordType`, etc) and call `super.setFlags()` fallowed by any rest logic for that class?

Reply:
Fixed in [83a25ae](https://github.com/ballerina-platform/ballerina-lang/pull/42777/commits/83a25ae0038f4ad4203317ad58aa0ed3b6e4481e)